src/r3/atomic-swap/corda/samples/atomic-swap/swap-workflows/src/main/kotlin/com/interop/flows/NotarizationSignaturesCollectorFlow.kt

```diff
@@ -0,0 +1,250 @@
+        val transactionId: SecureHash,
+        val transactionSignature: TransactionSignature,
+        val notary: PublicKey,
+        val blocking: Boolean
+    )
+
+    @CordaSerializable
+    data class StoreParams(
+        val transactionId: SecureHash,
+        val signature: ByteArray,
+        val notary: PublicKey,
+        val blocking: Boolean
+    ) {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as StoreParams
+
+            if (transactionId != other.transactionId) return false
+            if (!signature.contentEquals(other.signature)) return false
+            if (notary != other.notary) return false
+            if (blocking != other.blocking) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = transactionId.hashCode()
+            result = 31 * result + signature.contentHashCode()
+            result = 31 * result + notary.hashCode()
+            result = 31 * result + blocking.hashCode()
+            return result
+        }
+    }
+
+    @Suspendable
+    @StartableByRPC
+    @InitiatingFlow
+    class CollectNotarizationSignaturesFlow(
+        val transactionId: SecureHash,
+        val blocking: Boolean
+    ) : FlowLogic<Unit>() {
+
+        companion object {
+            val log = loggerFor<CollectNotarizationSignaturesFlow>()
+        }
+
+        @Suspendable
+        override fun call() {
+            // NOTE -> ME
+            val signedTransaction = serviceHub.validatedTransactions.getTransaction(transactionId)
+                ?: throw IllegalArgumentException("Transaction not found for ID: $transactionId")
+
+            val lockState = signedTransaction.tx.outputs
+                .mapNotNull { it.data as? LockState }
+                .singleOrNull()
+                ?: throw IllegalArgumentException("Transaction $transactionId does not have a lock state")
+
+            val validators = lockState.approvedValidators.mapNotNull {
+                serviceHub.identityService.partyFromKey(it)
+            }
+
+            val sessions = validators.map { initiateFlow(it) }
+
+            val notary = signedTransaction.tx.notary!!.owningKey
+
+            val signature = signedTransaction.sigs.singleOrNull {
+                it.by == notary
+            } ?: throw IllegalArgumentException("Transaction $transactionId is not signed by the expected notary")
+
+            for (session in sessions) {
+                try {
+                    session.send(
+                        RequestParams(
+                            transactionId,
+                            signature,
+                            notary,
+                            blocking
+                        )
+                    )
+                } catch (e: Throwable) {
+                    log.error("Error while sending request.\nError: $e")
```

Comment:
If we have an error here will that cause a problem in the receiving side below?  Does `session.receive()` timeout?  Or should you only `session.receive()` for sessions that were successfully sent to?